serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -66,7 +66,7 @@ class SequenceGenerationOutput:
     new_tokens: List[int]
     finish_reason: Optional[FinishReason] = None
     error: Optional[Union[str, ValidationError]] = None
-    logprob_info: Optional[RawLogprobsInfos] = None
+    logprob_info: Optional[List[Optional[RawLogprobsInfo]]] = None
```

Comment:
Same comment about types, these are crazy nestings its better to just split these into dataclasses, or name the types so we they have a logical identity

Reply:
Added to the ticket.